src/pg/mod.rs

```diff
@@ -367,10 +366,35 @@ impl AsyncPgConnection {
         //
         // We apply this workaround to prevent requiring all the diesel
         // serialization code to beeing async
-        let mut metadata_lookup = PgAsyncMetadataLookup::new();
+        let mut bind_collector_0 = RawBytesBindCollector::<diesel::pg::Pg>::new();
+        let collect_bind_result_0 = query.collect_binds(
```

Comment:
We want to have some comment here why these steps are necessary and how the approach works from a top level view. Just to make sure that future refactoring won't break it again.